### PR TITLE
Create Database class

### PR DESCRIPTION
**Pull request by dcarpenter31**
The original motivation for this change was to fix the getter database methods so that they returned `Items` instead of dictionaries. However, because these methods were part of `Item` and we needed the constructors for the `Item` subclasses in order to return a list of the correct items, we had a circular import error. This PR suggests that we create a `Database` class that is responsible for all database operations that we can use in the other types.

### CODE REVIEW

File: src/e4stypes/database.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+
+class Database:
+    @classmethod
+    def get_all(cls) -> List[Item]:
+        return (Database.get_item_by_category("Clothing") +
+                Database.get_item_by_category("Book") +
+                Database.get_item_by_category("Furniture") +
+                Database.get_item_by_category("Electronic") +
+                Database.get_item_by_category("Sports Gear"))
```
Comment (elenirotsides) [inline src/e4stypes/database.py:32]: Have you tested this `get_all` method? You might have to wrap each of these in a `list()` type conversion function for the data to be accessible, and for the data to return a List of Items (a list of dictionaries). By default, mongo will return a dictionary for this call`Database.get_item_by_category("Clothing")` for example, so to fix the return type, I would concat them and convert them to a list. I haven't tested this myself but I'm just wondering if you were able to get a result. @carpenterd777

File: src/e4stypes/database.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                                 item_dict["title"], item_dict["desc"],
+                                 Decimal(item_dict["price"]),
+                                 item_dict["weight"], item_dict["seller"],
+                                 item_dict["garment_type"],
+                                 ClothingSize(item_dict["size"]),
+                                 ClothingGender(item_dict["gender"]),
+                                 item_dict["color"]))
+            return items
+        elif category == 'Book':
+            item_dicts = book_col.find()
```
Comment (elenirotsides) [inline src/e4stypes/database.py]:
```suggestion
            item_dicts = book_col.find({})
```

Comment (dcarpenter31) [reply src/e4stypes/database.py]: this isnt necessary, but I'll add it back for consistency

**Review by elenirotsides — APPROVED** [review]
Looks good!

**Pull request merged**